### ISSUE TITLE
Tag support

### ISSUE THREAD

**Issue by Pinjasaur** [Pinjasaur/bic#10]
A major feature that would be worth considering support for would be arbitrary tags to help categorize posts.

High level scope:
- Determine the syntax to be used within the markdown document i.e. `tags: tag1, tag2, tag3`
- What URL structure for the index of tags along with all the documents tagged something i.e. `/tags` and either `/tags/{tag}` or `/tagged/{tag}`
- Roughly how the data structure would look internally to track tags to their source documents. Will certainly be janky because Bash doesn't make that trivial.

An initial proposal of a hypothetical markdown document:

```markdown
# My title
tags: foo, bar, baz

Here are some words.

Lorem ipsum.
```

I'm partial to `/tags` and `/tags/{tag}` since it's more of a RESTful syntax.

Open questions:
- How to parse tags? 
	- Split on commas, trim whitespace? Do we trim all whitespace? And lowercase?

**Comment by Pinjasaur** [comment]
cc @mfossen thoughts?

**Comment by Pinjasaur** [comment]
Going through some old notes I found some thoughts I had jotted down. Added a few more, too. But, I think technically speaking, this can be implemented using a `declare -A ALL_TAGS` to create an associative array data structure containing the tags as keys e.g. `mytag` and the values as a delimited string of posts e.g. `posts/foo.md:posts/bar.md`.

![image](https://user-images.githubusercontent.com/6335792/231305261-5eff235d-8611-469d-b8bc-b1355030f22b.png)
